app/views/locations/upload_summary.html.erb

```diff
@@ -0,0 +1,96 @@
+<h1 class="govuk-heading-l">Upload Summary</h1>
+
+<% if @parent_organisation.errors.present? %>
+  <div class="govuk-error-summary" role="alert" data-module="govuk-error-summary" aria-labelledby="error-summary-title">
+    <h2 id="error-summary-title" class="govuk-error-summary__title">
+      There is a problem
+    </h2>
+    <div class="govuk-error-summary__body">
+      <ul class="govuk-list govuk-error-summary__list">
+        <% @parent_organisation.locations.select(&:new_record?).each do |location| %>
+          <% location_errors = location.errors.where(:address).push(*location.errors.where(:postcode)) %>
+          <% location_errors.each_with_index do | error, i | %>
+            <li>
+              <a href='#<%= upload_error_id_helper(i, location.address) %>'>
+                <%= error.full_message %>
+              </a>
+            </li>
+          <% end %>
+          <% location.ips.each do |ip| %>
+              <% ip.errors.full_messages.each_with_index do | error_message, i | %>
+              <li>
+                <a href='#<%= upload_error_id_helper(i, location.address, ip.address) %>'>
+                  <%= error_message %>
+                </a>
+              </li>
+            <% end %>
+          <% end %>
+        <% end %>
+      </ul>
+    </div>
+  </div>
+<% end %>
+
+<% @parent_organisation.locations.select(&:new_record?).each do |location| %>
+
+<fieldset class="govuk-fieldset">
+  <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
+    <% location_errors = location.errors.where(:address).push(*location.errors.where(:postcode)) %>
+    <% if location_errors.any? %>
+      <div class="govuk-form-group govuk-form-group--error">
+        <% location_errors.each_with_index do | error, i | %>
+          <p class="govuk-error-message" id='<%= upload_error_id_helper(i, location.address) %>'>
+              <span class="govuk-visually-hidden">
+                Error:
+              </span>
+            <%= error.full_message %>
+          </p>
+        <% end %>
+        <h1 class="govuk-fieldset__heading">
+          <%= location.address %>, <%= location.postcode %>
+        </h1>
+      </div>
+    <% else %>
+      <h1 class="govuk-fieldset__heading">
+        <%= location.address %>, <%= location.postcode %>
+      </h1>
+    <% end %>
+  </legend>
+
+  <% location.ips.each do |ip| %>
+    <% if ip.errors.any? %>
+      <% ip.errors.full_messages.each_with_index do |error_message, i| %>
```

Comment:
If you have more than one validation error, it will seem that there are double the number of IP addresses.

I think the design system says you only display the first?